apps/next/app/dashboard/page.tsx

```diff
@@ -45,30 +46,49 @@ async function findCreateUserWithClerkId({
   return newUser;
 }
 
+const ensureTeamForClerkOrg = async (): Promise<Team | undefined> => {
+  /*
+  - If a user is already has a Metal account and then uses Clerk's <OrganizationSwitcher />
+  to create a new org, the new Clerk org will exist but won't have an associated Metal team. This
+  function will ensure that none of the user's prior teams are not matched, thus
+  signaling to clients of this function that they need to create a new team.
+  - If a user is newly registered, they won't have any organizations or teams. This
+  function will detect that too and return no team.
+   */
+  const { orgId, userId } = auth();
+  if (orgId) {
+    const usersPersonalTeam = await db.query.teams
+      .findMany({
+        where: (team, { eq, and }) =>
+          and(eq(team.clerkId, orgId), eq(team.creatorId, userId)),
+      })
+      .then((rows) => rows[0] || undefined);
+
+    return usersPersonalTeam;
+  }
+};
```

Comment:
this is a really confusing function...

it's performing a search for a team in our db that is created by the logged in user and matches the session's active clerk org. I am struggling mightily to think about the situations where this search would be useful... is it just for the moment right after a user creates a new organization?

I also question whether this function even works? `creatorId` is an id from our database, and `userId` is a clerk id

Reply:
I get this error after logging in 

![image](https://github.com/onmetal-dev/metal/assets/72655/6ed708f2-7835-4443-8299-fc9dae5c78f4)

Reply:
This should be fixed now. LMK if the problem persists.